tests/test_runner_apps/tagged/tests_syntax_error.py

```diff
@@ -0,0 +1,11 @@
+from unittest import TestCase
+
+from django.test import tag
+
+
+@tag('syntax_error')
+class SyntaxErrorTestCase(TestCase):
+    pass
+
+
+syntax_error := 1  # NOQA
```

Comment:
Isn't this valid in Python 3.8 though? That's the problem with using the correct (matching) tag in the test. It will still pass even without a syntax error.

Reply:
The walrus operator is not valid at the top level:
```python
>>> a := 1
  File "<stdin>", line 1
    a := 1
      ^
SyntaxError: invalid syntax
```
I've switched to sth more obvious.

Reply:
Okay, thanks.

Reply:
> That's the problem with using the correct (matching) tag in the test. It will still pass even without a syntax error.

Are you sure? Maybe I'm missing sth, but `test_tag_fail_to_load` fails (as expected) without the fix :thinking: :
```
FAIL: test_tag_fail_to_load (test_runner.test_discover_runner.DiscoverRunnerTests)
----------------------------------------------------------------------
....
AssertionError: 0 != 2
```

Reply:
> It will still pass even without a syntax error.

I just meant that it would still pass (with the fix) if the test module raised e.g. `NameError` instead of `SyntaxError`. (I also thought it would pass if the test module raised no error, but I see now that's not the case since `SyntaxErrorTestCase` doesn't contain a test.)

In any case, I would do something like the following because it checks both that (1) the test module really does raise a syntax error, and (2) the tests you're picking up are in fact the ones coming from the syntax error module and the module that isn't found (as opposed to potentially other test modules in that test directory):

```python
    def test_tag_fail_to_load(self):
        with self.assertRaises(SyntaxError):
            importlib.import_module('test_runner_apps.tagged.tests_syntax_error')
        runner = DiscoverRunner(tags=['syntax_error'])
        # A label that doesn't exist or cannot be loaded due to syntax errors
        # is always considered matching.
        suite = runner.build_suite(['doesnotexist', 'test_runner_apps.tagged'])
        self.assertEqual(suite.countTestCases(), 2)
        self.assertEqual([test.id() for test in suite], [
            'unittest.loader._FailedTest.doesnotexist',
            'unittest.loader._FailedTest.test_runner_apps.tagged.tests_syntax_error',
        ])
```

Reply:
Agreed, thanks for the suggestion :star:

Reply:
Np, thanks again for all your help, @felixxm.